corehq/form_processor/tests/test_basics.py

```diff
@@ -447,6 +449,34 @@ def test_globally_unique_case_id(self):
             self.assertTrue(xform.is_error)
             self.assertIn('IllegalCaseId', xform.problem)
 
+    def test_duplicate_with_attachment(self):
+        def main():
+            attachments = {
+                'pic.jpg': UploadedFile(BytesIO(b"fake"), 'pic.jpg', content_type='image/jpeg', size=4)
+            }
+            form_id = uuid.uuid4().hex
+            form = get_simple_form_xml(form_id)
+
+            form_1 = submit_and_fetch(form, DOMAIN, attachments)
+            self.assert_(not form_1.is_duplicate)
+            check_attachments(form_1, attachments)
+
+            form_1 = self.formdb.get_form(form_id)
+            form_2 = submit_and_fetch(form, DOMAIN, attachments)
+            self.assert_(not form_1.is_duplicate)
```

Comment:
It looks like `form_1` is stale by the time this assertion is done. Would it make sense to fetch `form_1` after submitting `form_2` (i.e., swap line 464 and 465)?

Reply:
Great point, thank you: https://github.com/dimagi/commcare-hq/pull/26895